src/main/scala/it/pagopa/interop/agreementmanagement/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -98,6 +98,36 @@ final case class AgreementApiServiceImpl(
   private def createAgreement(agreement: PersistentAgreement): Future[StatusReply[PersistentAgreement]] =
     commander(agreement.id.toString).ask(ref => AddAgreement(agreement, ref))
 
+  /**
+   * Code: 204, Message: Agreement deleted
+   * Code: 400, Message: Invalid input, DataType: Problem
+   * Code: 404, Message: Agreement not found, DataType: Problem
+   */
+  override def deleteAgreement(
+    agreementId: String
+  )(implicit toEntityMarshallerProblem: ToEntityMarshaller[Problem], contexts: Seq[(String, String)]): Route =
+    authorize(ADMIN_ROLE) {
+      val operationLabel: String = "Deleting agreement"
```

Comment:
the `agreementId` is basically always used together with this label, it could be refactored to 
```suggestion
      val operationLabel: String = s"Deleting agreement $agreementId"
```